content/en/docs/Extensions/apiKeyAuth.md

```diff
@@ -0,0 +1,47 @@
+---
+title: "API Key Authentication"
+linkTitle: "API Key Authentication"
+OpenSubsonic:
+  - Extension
+description: >
+  Add a new authentication mechanism involving only an API key, and no
+---
+
+**OpenSubsonic version**: [1](../../opensubsonic-versions)
+
+**OpenSubsonic extension name** `apiKeyAuthentication` (As returned by [`getOpenSubsonicExtensions`](../../endpoints/getopensubsonicextensions))
+
+## Version 1
+
+This extension requires changes to the semantics of authentication.
+Broadly, there are two general changes:
+
+1. **Required**: A new authentication mechanism: `apiKey` for query.
+2. Recommended: Deprecation of token/salt-based authentication.
+
+### API keys
+
+An API key is any authentication token generated by an OpenSubsonic server that can be used to authenticate.
+How this API key is generated by the server is implementation-specific: the server may provide a page where the user can configure one or more API keys, the server may automatically generate a API key, or any other variety of means.
+The format of the API key is not specified, but it **must** be of reasonable length to fit into a query parameter (less than 2048 characters URL-encoded).
+
+Servers which implement this extension **must** provide some mechanism for viewing active API key(s) and allow for revoking API keys.
+Note that these API keys **do not** expire; as long as they are not revoked by the user, they are assumed to be valid.
+
+#### Using a API key
+
+An API key is used as a query parameter `apiKey=<api key>`.
+When an API key is provided, the client **must not** provide a `u` parameter; passing in `u` **must** be treated as an error `43`.
+
+It is **recommended** that servers which provide API-key authentication deprecate salt/token-based authentication.
+
+If multiple conflicting authentication parameters are passed in, the server **must** return an error `43`, `Multiple conflicting authentication mechanisms provided`
+
+If a server deprecates salt-based authentication, it **must** return an error `41` (`Token authentication not supported for LDAP users.`).
```

Comment:
This is quite minor but I think that this now mixes up the terminology a bit. As I have understood it, "deprecated" feature is something which is still supported but discouraged and destined to be removed sometime in the future. But if we say that an error must be returned, then the feature is not merely deprecated but outright removed.

Reply:
Yeah, more correct to say removes support.